Lib/test/test_pty.py

```diff
@@ -51,14 +51,11 @@ class PtyTest(unittest.TestCase):
     def setUp(self):
         # isatty() and close() can hang on some platforms.  Set an alarm
         # before running the test to make sure we don't hang forever.
-        self.old_alarm = signal.signal(signal.SIGALRM, self.handle_sig)
+        old_alarm = signal.signal(signal.SIGALRM, self.handle_sig)
+        self.addCleanup(signal.signal, signal.SIGALRM, old_alarm)
```

Comment:
Why this is changed?

Reply:
In my experience, tearDown() is an called if a test raises an exception. But I'm not sure about that.

Well, I also like the addCleanup() API to move the cleanup code closer to the setup code.